python/lsst/ts/m2gui/controltab/tab_alarm_warn.py

```diff
@@ -135,6 +148,63 @@ def _set_text_error_cause(self):
 
         return text_error_cause
 
+    def _set_indicators_limit_switch(self, limit_switch_type):
+        """Set the indicators of limit switch.
+
+        Parameters
+        ----------
+        limit_switch_type : enum `LimitSwitchType`
+            Type of limit switch.
+
+        Returns
+        -------
+        indicators : `dict`
+            Indicators of limit switch.
+
+        Raises
+        ------
+        ValueError
+            Unsupported limit switch type.
+        """
+
+        if limit_switch_type == LimitSwitchType.Retract:
+            limit_switch_status = self.model.fault_manager.limit_switch_status_retract
+        elif limit_switch_type == LimitSwitchType.Extend:
+            limit_switch_status = self.model.fault_manager.limit_switch_status_extend
+        else:
+            raise ValueError(f"Unsupported limit switch type: {limit_switch_type!r}.")
+
+        indicators = dict()
+        for name in limit_switch_status.keys():
+            indicator = set_button(name, None, is_indicator=True)
+            indicator.adjustSize()
```

Comment:
Is that needed? AdjustSize shall be called in constructor/after setting button text..

Reply:
Updated in the commit: 0c06a5b. Thanks!